### ISSUE TITLE
UCX failed to built in Fedora 30 (GCC 9.0.1) — v1.5.0

### ISSUE THREAD

**Issue by amaslenn** [openucx/ucx#3289]
Fedora 30 logs (failed): https://koji.fedoraproject.org/koji/taskinfo?taskID=33143707
Fedora 31 logs (failed): https://koji.fedoraproject.org/koji/taskinfo?taskID=33143693
Fedora 29 logs (passed, gcc 8.3): https://koji.fedoraproject.org/koji/taskinfo?taskID=33143720

Output ([log](https://kojipkgs.fedoraproject.org//work/tasks/3695/33143695/build.log)):
```sh
...
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -DCPU_FLAGS= -I/builddir/build/BUILD/ucx-1.5.0/src -I/builddir/build/BUILD/ucx-1.5.0/src -DLIB_MLX5_VER=\"\" -O3 -g -Wall -Werror -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -c ib/ud/base/ud_iface_common.c  -fPIC -DPIC -o ib/ud/base/.libs/libuct_la-ud_iface_common.o
BUILDSTDERR: In file included from /builddir/build/BUILD/ucx-1.5.0/src/ucs/sys/sys.h:16,
BUILDSTDERR:                  from /builddir/build/BUILD/ucx-1.5.0/src/uct/base/uct_iface.h:18,
BUILDSTDERR:                  from /builddir/build/BUILD/ucx-1.5.0/src/uct/ib/rc/base/rc_iface.h:12,
BUILDSTDERR:                  from /builddir/build/BUILD/ucx-1.5.0/src/uct/ib/dc/base/dc_iface.h:10,
BUILDSTDERR:                  from ib/dc/accel/dc_mlx5.h:10,
BUILDSTDERR:                  from ib/dc/accel/dc_mlx5.c:7:
BUILDSTDERR: ib/dc/accel/dc_mlx5.c: In function 'uct_dc_mlx5_ep_fc_ctrl':
BUILDSTDERR: ib/dc/accel/dc_mlx5.c:1019:70: error: taking address of packed member of 'struct <anonymous>' may result in an unaligned pointer value [-Werror=address-of-packed-member]
BUILDSTDERR:  1019 |                                                    ucs_unaligned_ptr(&dc_req->sender.global.gid),
BUILDSTDERR: /builddir/build/BUILD/ucx-1.5.0/src/ucs/sys/compiler.h:71:42: note: in definition of macro 'ucs_unaligned_ptr'
BUILDSTDERR:    71 | #define ucs_unaligned_ptr(_ptr) ((void*)(_ptr))
BUILDSTDERR:       |                                          ^~~~
...
```

**Comment by amaslenn** [comment]
New fails:
```sh
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -DCPU_FLAGS= -I/builddir/build/BUILD/ucx-1.5.0/src -I/builddir/build/BUILD/ucx-1.5.0/src -DLIB_MLX5_VER=\"\" -O3 -g -Wall -Werror -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -c sm/mm/mm_md.c -o sm/mm/libuct_la-mm_md.o >/dev/null 2>&1
BUILDSTDERR: ib/ud/accel/ud_mlx5.c: In function 'uct_ud_mlx5_iface_poll_rx':
BUILDSTDERR: ib/ud/accel/ud_mlx5.c:423:26: error: converting a packed 'uct_ib_iface_recv_desc_t' {aka 'struct uct_ib_iface_recv_desc'} pointer (alignment 1) to a 'uct_ud_recv_skb_t' {aka 'struct uct_ud_recv_skb'} pointer (alignment 8) may result in an unaligned pointer value [-Werror=address-of-packed-member]
BUILDSTDERR:   423 |                          (uct_ud_recv_skb_t *)desc, is_async);
BUILDSTDERR:       |                          ^
BUILDSTDERR: In file included from /builddir/build/BUILD/ucx-1.5.0/src/uct/ib/base/ib_log.h:11,
BUILDSTDERR:                  from /builddir/build/BUILD/ucx-1.5.0/src/uct/ib/mlx5/ib_mlx5.h:13,
BUILDSTDERR:                  from ib/ud/accel/ud_mlx5_common.h:10,
BUILDSTDERR:                  from ib/ud/accel/ud_mlx5.h:9,
BUILDSTDERR:                  from ib/ud/accel/ud_mlx5.c:8:
BUILDSTDERR: /builddir/build/BUILD/ucx-1.5.0/src/uct/ib/base/ib_iface.h:273:16: note: defined here
BUILDSTDERR:   273 | typedef struct uct_ib_iface_recv_desc {
BUILDSTDERR:       |                ^~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR: In file included from /builddir/build/BUILD/ucx-1.5.0/src/uct/ib/ud/base/ud_iface.h:22,
BUILDSTDERR:                  from ib/ud/accel/ud_mlx5.h:11,
BUILDSTDERR:                  from ib/ud/accel/ud_mlx5.c:8:
BUILDSTDERR: /builddir/build/BUILD/ucx-1.5.0/src/uct/ib/ud/base/ud_def.h:180:16: note: defined here
BUILDSTDERR:   180 | typedef struct uct_ud_recv_skb {
BUILDSTDERR:       |                ^~~~~~~~~~~~~~~
```

Full log: https://kojipkgs.fedoraproject.org//work/tasks/5414/33175414/build.log

**Comment by amaslenn** [comment]
Looks good now: https://koji.fedoraproject.org/koji/taskinfo?taskID=33223406 👍
